### ISSUE TITLE
Gray out unnecessary non-null suppression operator

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#26736]
**Version Used**: Nullable Reference Types Preview (7 May 2018)

**Steps to Reproduce**:

```csharp
object? obj = new object();
var str = obj!.ToString();
```

**Expected Behavior**:

A warning is provided for the use of `!` where flow analysis already determines the expression cannot be null.

**Actual Behavior**:

No warning is provided.

**Comment by alrz** [comment]
Conversely https://github.com/dotnet/roslyn/issues/22743

**Comment by jcouv** [comment]
Last LDM discussion on this landed on:
> These constructs are meaningless, but unlikely to do something the user didn't want. No compiler warnings for !!... or nonNull!. Maybe an IDE feature.

https://github.com/dotnet/csharplang/blob/master/meetings/2018/LDM-2018-08-20.md#12-meaningless--operators

**Comment by jcouv** [comment]
Updated area to IDE and title to "gray out unnecessary suppression".

**Comment by jasonmalinowski** [comment]
@jcouv @333fred If the IDE _did_ want to implement this, what would be the recommended way to do so? We'd just call GetTypeInfo on the thing to the left and if it's NullableFlowState.NotNull then flag it? Or is there more subtlety needed here?

(We're keeing this in backlog for now but just curious if this is something we even could do or if I should move it to 'blocked by compiler API')

**Comment by 333fred** [comment]
More subtlety will be needed here. The suppression operator is going to change what we report for `GetTypeInfo`, so `NotNull` will be reported for this. I think this will likely be blocked by https://github.com/dotnet/roslyn/issues/35412.

**Comment by jcouv** [comment]
Definitely more subtle. The suppression operator also conversions and nested nullability. For example, `List<string> x = new List<string?>()!; // warning is suppressed by ! so is "necessary"`.

**Comment by sharwell** [comment]
This is blocked on #25372

**Comment by sharwell** [comment]
Duplicate of #34714
